setup.py

```diff
@@ -1,9 +1,23 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
+import os
+import pkgutil
 import re
+import shutil
+import sys
+from codecs import open
 
 from setuptools import setup, find_packages
-from codecs import open
+from setuptools.command.develop import develop
+from setuptools.command.install import install
+
+try:
+    import bs4
```

Comment:
Requiring `bs4` is a bit too heavy. What if the `modular_template.html` is still vendored manually in mesa-geo?

Reply:
There has been a number of changes to `modular_template.html`. Someone will need to keep an eye on it and update accordingly if it's included manually.

I put `bs4` into `pyproject.toml` file separately, hopefully as a reminder that it's only needed at build time. As soon as https://github.com/projectmesa/mesa/issues/1267 is taken care of, `bs4` can be removed.

Reply:
If `bs4` is a temporary solution, vendoring `modular_template.html` should also be viable as a temporary solution. It shouldn't take that long to implement https://github.com/projectmesa/mesa/issues/1267.

Reply:
ok : ) it's brought back and the changes are squashed into the first commit.